WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergGalleryUploadProcessor.swift

```diff
@@ -0,0 +1,143 @@
+            case let .string(imgClass) = imgClassAttributeValue else {
+                return nil
+        }
+
+        let classAttributes = imgClass.components(separatedBy: " ")
+
+        guard let imageIDAttribute = classAttributes.filter({ (value) -> Bool in
+            value.hasPrefix(GutenbergImgUploadProcessor.imgClassIDPrefixAttribute)
+        }).first else {
+            return nil
+        }
+
+        let imageIDString = String(imageIDAttribute.dropFirst(ImageKeys.classIDPrefix.count))
+        let imgUploadID = Int32(imageIDString)
+
+        guard imgUploadID == self.mediaUploadID else {
+            return nil
+        }
+
+        let newImgClassAttributes = imgClass.replacingOccurrences(of: imageIDAttribute, with: ImageKeys.classIDPrefix + String(self.serverMediaID))
+
+        var attributes = img.attributes
+        attributes.set(.string(self.remoteURLString), forKey: "src")
+        attributes.set(.string(newImgClassAttributes), forKey: "class")
+        attributes.set(.string("\(self.serverMediaID)"), forKey: ImageKeys.dataID)
+        attributes.set(.string(self.remoteURLString), forKey: ImageKeys.dataFullURL)
+        if attributes.contains(where: { $0.key == ImageKeys.dataLink } ) {
+            attributes.set(.string(self.mediaLink), forKey: ImageKeys.dataLink)
+        }
+
+        var html = "<\(ImageKeys.name) "
+        let attributeSerializer = ShortcodeAttributeSerializer()
+        html += attributeSerializer.serialize(attributes)
+        html += " />"
+        return html
+    })
+
+    private struct LinkKeys {
+        static let name = "a"
+    }
+
+    lazy var linkPostMediaUploadProcessor = HTMLProcessor(for: LinkKeys.name, replacer: { (link) in
+
+        guard let linkOriginalContent = link.content else {
+            return nil
+        }
+
+        let linkUpdatedContent = self.imgPostMediaUploadProcessor.process(linkOriginalContent)
+
+        guard linkUpdatedContent != linkOriginalContent else {
+            return nil
+        }
+
+        var attributes = link.attributes
+        if let linkToURL = self.linkToURL {
+            attributes.set(.string(linkToURL), forKey: "href")
+        } else {
+            attributes.set(.string(self.remoteURLString), forKey: "href")
+        }
+
+        var html = "<\(LinkKeys.name) "
+        let attributeSerializer = ShortcodeAttributeSerializer()
+        html += attributeSerializer.serialize(attributes)
+        html += " >"
+        html += linkUpdatedContent
+        html += "</\(LinkKeys.name)>"
+        return html
+    })
+
+    private struct GalleryBlockKeys {
+        static let name = "wp:gallery"
+        static let ids = "ids"
+        static let linkTo = "linkTo"
+    }
+
+    lazy var galleryBlockProcessor = GutenbergBlockProcessor(for: GalleryBlockKeys.name, replacer: { block in
+        guard var ids = block.attributes[GalleryBlockKeys.ids] as? [Int32],
```

Comment:
There's a small problem here, when there are already pictures in the gallery some ids are coming as Strings, so the execution is not going any further than this.

<img width="613" alt="Screen Shot 2019-12-12 at 17 56 51" src="https://user-images.githubusercontent.com/5032900/70723458-f9bc4d80-1d09-11ea-9e8a-fd86c0d7a6a6.png">

Steps:
- Create a gallery block with images from WP media library
- Save & Close post
- Reopen post
- Start uploading image
- Close post, wait until uplooads end
- Open the post again
- Result: Gallery still has temp ids and urls

I think it'd be good if we can also add a unittest for this scenario.

Reply:
I changed the code to be able two support String or Ints or both on the array of ids and also added unit tests to check all the possible scenarios.

You can see the fixes here: https://github.com/wordpress-mobile/WordPress-iOS/pull/13032/commits/6d05a199aa0838ca0cecb13a33a7fa15114d9e09